### ISSUE TITLE
FunctionDeclarationArgumentSpacingSniff works space after type hint

### ISSUE THREAD

**Issue by webimpress** [squizlabs/PHP_CodeSniffer#2284]
The `FunctionDeclarationArgumentSpacingSniff` works incorrectly for more complex type hints like:

```php
function (
    ?array    $arr,
    \My\Name    $name
) {}
```

It should be easy fix, but I don't have time to do it now...

**Comment by webimpress** [comment]
I was mistaken here, the problem is because PSR-2 has:

```xml
<rule ref="Squiz.Functions.FunctionDeclarationArgumentSpacing.SpacingAfterHint">
  <severity>0</severity>
 </rule>
```

but I have found the other problem with above code, it was fixed as:

```php
function (
    ? $arr,
    \My\Name $name
) {}
```

so `array` has been removed. 

I'm going to submit PR in a minute.
